### DISCUSSION TITLE
How get/execute model by name:string

### DISCUSSION THREAD

**Discussion by eliasrosa** [prisma/prisma#6061] [Q&A]
Hello,

I am creating a function to standardize a simple pagination.

```ts
import db, { Prisma } from '@database/connection';

//
export interface IPaginateResponse {
  page?: number;
  take?: number;
  sort?: string;
  search?: string;
  data?: unknown[];
  total: number;
}

type Model = Uncapitalize<keyof typeof Prisma.ModelName>;

const Paginate = async (model: Model): Promise<IPaginateResponse> => {
  // get model
  const m = db[model];

  // get count result
  const total = await m.count();

  // return empty on not result count
  if (!total) {
    // return []
  }

  return {
    total,
  };
};

export default Paginate;
```

By typing down I managed to pull the models from the prisma.

```ts
type Model = Uncapitalize<keyof typeof Prisma.ModelName>;
```

![image](https://user-images.githubusercontent.com/84581/110800929-787c9f00-825b-11eb-9719-fedb037f5ef2.png)

Basically, the first thing to do is to count the records, OK!
The count is working and returned, but the typescript is giving the following error:

```txt
Count the number of Users. Note, that providing undefined is treated as the value not being there. Read more here: https://pris.ly/d/null-undefined Count the number of PriceLists. Note, that providing undefined is treated as the value not bein...
```
![image](https://user-images.githubusercontent.com/84581/110803152-a82ca680-825d-11eb-8d46-d8f55819c870.png)

Any idea how to solve?

Thank you very much!

**Prisma version:**
prisma : 2.18.0
@prisma/client  : 2.18.0

**Comment by ryands17** [comment]
@eliasrosa 👋 

Currently this approach is not recommended in TypeScript as you wouldn't be able to exactly type the model and its arguments along with the return type. There's a feature request for the same with some possible solutions presented: https://github.com/prisma/prisma/issues/5273

I would suggest going through them and adding a 👍 to this so that you can follow this request :)

**Reply by eliasrosa** [reply]
Thank you very much for the quick response,

if you find any alternative, I'll post it here!

tks

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
